ui/src/main.rs

```diff
@@ -72,10 +79,45 @@ fn main() {
     let logger = StatisticLogger::new(file_logger);
     chain.link_around(logger);
 
+    if env::var("CORS").is_ok() {
+        debug!("using CORS");
+        chain.link_around(CorsMiddleware);
+        chain.link_after(CorsMiddleware);
+    }
+
     info!("Starting the server on {}:{}", address, port);
     Iron::new(chain).http((&*address, port)).expect("Unable to start server");
 }
 
+struct CorsMiddleware;
+
+impl AroundMiddleware for CorsMiddleware {
+    fn around(self, handler: Box<Handler>) -> Box<Handler> {
+        Box::new(move | req: &mut Request | {
+            match req.method {
+                Method::Options => Ok(Response::with((status::Ok))),
```

Comment:
Yes, you're right. Ideally one would also check if the route is already handling I can add a guard here to only respond to request with such headers. It might be worthwhile to not respond ahead of time for routes who are already OPTION requests, but I am unsure if it's possible to identify those at this stage. I also noticed that you are using Mount and not Router, so I assume that all routes accept all HTTP methods? I